File: GeoBroker-Commons/src/main/java/de/hasenburg/geobroker/commons/model/message/payloads/DISCONNECTPayload.java
package de.hasenburg.geobroker.commons.model.message.payloads;

import de.hasenburg.geobroker.commons.model.BrokerInfo;
import de.hasenburg.geobroker.commons.model.message.ReasonCode;

import java.util.Objects;

public class DISCONNECTPayload extends AbstractPayload {

	private ReasonCode reasonCode;
	private BrokerInfo brokerInfo;

	public DISCONNECTPayload() {

	}

	public DISCONNECTPayload(ReasonCode reasonCode) {
		super();
		this.reasonCode = reasonCode;
	}

	public DISCONNECTPayload(ReasonCode reasonCode, BrokerInfo brokerInfo) {
		super();
		this.reasonCode = reasonCode;
		this.brokerInfo = brokerInfo;
	}

	// overwrite as BrokerInfo might be null
	public boolean nullField() {
		return reasonCode == null;
	}

	/*****************************************************************
	 * Getter & Setter
	 ****************************************************************/

	public ReasonCode getReasonCode() {
		return reasonCode;
	}

	public BrokerInfo getBrokerInfo() {
		return brokerInfo;
	}

	/*****************************************************************
	 * Generated Code
	 ****************************************************************/

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		DISCONNECTPayload that = (DISCONNECTPayload) o;
		return reasonCode == that.reasonCode && Objects.equals(brokerInfo, that.brokerInfo);
	}

	@Override
	public int hashCode() {
		return Objects.hash(reasonCode, brokerInfo);
	}

}

File: GeoBroker-Commons/src/main/kotlin/de/hasenburg/geobroker/commons/model/message/payloads/DISCONNECTPayload.kt
package de.hasenburg.geobroker.commons.model.message.payloads

import de.hasenburg.geobroker.commons.model.BrokerInfo
import de.hasenburg.geobroker.commons.model.message.ReasonCode

import java.util.Objects

class DISCONNECTPayload(val reasonCode: ReasonCode) : AbstractPayload() {

    var brokerInfo: BrokerInfo? = null

    constructor(reasonCode: ReasonCode, brokerInfo: BrokerInfo?) : this(reasonCode) {
        this.brokerInfo = brokerInfo
    }

    /*****************************************************************
     * Generated methods
     ****************************************************************/

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as DISCONNECTPayload

        if (reasonCode != other.reasonCode) return false
        if (brokerInfo != other.brokerInfo) return false

        return true
    }

    override fun hashCode(): Int {
        var result = reasonCode.hashCode()
        result = 31 * result + (brokerInfo?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "DISCONNECTPayload(reasonCode=$reasonCode, brokerInfo=$brokerInfo)"
    }

}
